setup

```diff
@@ -0,0 +1,84 @@
+whoisthis="Eric Harrison"  #your name, used for Git config
+youremail="ericmharrison@gmail.com"  #your email, used for Git config
+
+#update package list and upgrade all software
+sudo apt-get update 2>/dev/null && sudo apt-get -fy full-upgrade
+sudo apt -fy install software-properties-common apt-transport-https wget
+
+#check for git
+hasgit=$(apt list git --installed 2>/dev/null)
+if [ "$hasgit" == "Listing..." ]
+then
+  echo "Git's not installed! Let's fix that!"
+  sudo apt-get -fy install git
+fi
+
+#congratulate yourself
+echo
+echo "Git is installed: $(git --version)"
+
+#configure git
+git config --global --replace-all user.name "$whoisthis"
+git config --global user.email "$youremail"
+echo
+echo "Git User Name: $(git config --global --get user.name)"
+echo "Git User Email: $(git config --global --get user.email)"
+
+#add keys for new repositories
+echo "Adding keys"
+while read key
```

Comment:
Are these key files generated by installers?

Reply:
they're public keys for microsoft and google for their repositories.  i have to look back at that to see if it's actually needed